### ISSUE TITLE
Is there a more convenient way to exclude sniffs?

### ISSUE THREAD

**Issue by Solvent13** [squizlabs/PHP_CodeSniffer#3431]
By default ```phpcs -e``` outputs that the Squiz code style has 15 sniffs. For example I want to exclude:

- Squiz.WhiteSpace.ControlStructureSpacing
- Squiz.WhiteSpace.ScopeClosingBrace
- Squiz.Functions.MultiLineFunctionDeclaration

And so logically in the xml settings file I specify:
```
<rule ref="Squiz">
        <exclude name="Squiz.WhiteSpace.ControlStructureSpacing"/>
        <exclude name="Squiz.WhiteSpace.ScopeClosingBrace"/>
        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration"/>
</rule>
```
But it causes problems: instead of the 15 rules of Squiz the ```phpcs -e``` command reports the Squiz code style already has 98 sniffs. Instead of eliminating the three rules, more rules were included in the checking.

In this situation, the working way is to exclude each rule individually:
```
<rule ref="Squiz.WhiteSpace.ControlStructureSpacing">
        <exclude name="Squiz.WhiteSpace.ControlStructureSpacing"/>
</rule>

<rule ref="Squiz.WhiteSpace.ScopeClosingBrace">
        <exclude name="Squiz.WhiteSpace.ScopeClosingBrace"/>
</rule>

<rule ref="Squiz.Functions.MultiLineFunctionDeclaration">
        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration"/>
</rule>
```
But it is cumbersome and inconvenient. So the question in the title is.

**Comment by jrfnl** [comment]
@Solvent13 Based on your question, there are a number of misunderstandings in play.

1. Using `phpcs -e` will report the list of currently applicable sniffs. If there is a custom ruleset (`phpcs.xml[.dist]` file) in place, it will use that, if not, it defaults to the `PEAR` standard. I suspect you already had a custom ruleset in place and that that ruleset set your standard to `PSR12`.
2. If you use the `phpcs -e` command with the `--standard=...` argument, you can confirm this.
    `phpcs -e --standard=PEAR` shows a total of 28 sniffs, with one sniff coming from the `Squiz` collection.
    `phpcs -e --standard=PSR12` shows a total of 59 sniffs, with 15 sniffs coming from the `Squiz` collection.
    `phpcs -e --standard=Squiz` shows a total of 133 sniffs, with 98 sniffs coming from the `Squiz` collection.
3. Exclusion works based on the applicable ruleset.
    So if you do the below, you are first **including** the complete `Squiz` ruleset (on top of what's already in your custom ruleset) and then excluding three sniffs from that standard.
    ```xml
    <rule ref="Squiz">
        <exclude name="Squiz.WhiteSpace.ControlStructureSpacing"/>
        <exclude name="Squiz.WhiteSpace.ScopeClosingBrace"/>
        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration"/>
    </rule>
    ```
    What I suspect you _want_ to do, is _exclude based on the existing ruleset in place_. If that's `PSR12` as I suspect, that would look like this:
    ```xml
    <rule ref="PSR12">
        <exclude name="Squiz.WhiteSpace.ControlStructureSpacing"/>
        <exclude name="Squiz.WhiteSpace.ScopeClosingBrace"/>
        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration"/>
    </rule>
    ```

In other words, this sounds like user error.

To learn more about how to configure a custom ruleset correctly, please read through the [examples in the wiki](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset).
